### PR TITLE
hotfix/Fix_Unit_tests_for_HMIMessageHandler_in_genivi

### PR DESCRIPTION
**Pull request by OHerasym**
Added hmi_message_handler unit tests.

Related:
        APPLINK-19074

### CODE REVIEW

**Comment by monstersem** [comment]
Please review: @Kozoriz, @iAndrew5 , @VVeremjova , @pestOO

File: src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  virtual void TearDown() OVERRIDE {
+    hmi_handler_->set_message_observer(NULL);
+    hmi_message_handler::MockHMIMessageObserver::destroy();
+    hmi_message_handler::HMIMessageHandlerImpl::destroy();
+    delete mb_adapter_;
+  }
+};
+
+TEST_F(HMIMessageHandlerImplTest,
+       OnErrorSending_EmptyMessage_ExpectOnErrorSendingProceeded) {
```
Comment (Kozoriz) [inline src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc]: *expect* is extra

**Comment by Kozoriz** [comment]
reviewed

File: src/components/hmi_message_handler/src/hmi_message_handler_impl.cc [hunk 2]
```diff
@@ -89,15 +89,19 @@ void HMIMessageHandlerImpl::OnErrorSending(MessageSharedPointer message) {
...
   message_adapters_.insert(adapter);
 }
 
 void HMIMessageHandlerImpl::RemoveHMIMessageAdapter(
     HMIMessageAdapter* adapter) {
   LOG4CXX_AUTO_TRACE(logger_);
-  if (adapter != NULL) {
-    message_adapters_.erase(adapter);
+  if (!adapter) {
+    return;
```
Comment (pestOO) [inline src/components/hmi_message_handler/src/hmi_message_handler_impl.cc]: May be add warnign log?

File: src/components/hmi_message_handler/include/hmi_message_handler/mock_hmi_message_observer.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    public utils::Singleton<MockHMIMessageObserver> {
+ public:
+  MOCK_METHOD1(OnMessageReceived,
+      void(utils::SharedPtr<application_manager::Message> message));
+  MOCK_METHOD1(OnErrorSending,
+      void(utils::SharedPtr<application_manager::Message> message));
+  virtual ~MockHMIMessageObserver() {}
+};
+
+}// namespace hmi_message_handler
```
Comment (iAndrew5) [inline src/components/hmi_message_handler/include/hmi_message_handler/mock_hmi_message_observer.h]: 2 spaces between code and comments everywhere in files

**Comment by VVeremjova** [comment]
Reviewed

File: src/components/hmi_message_handler/test/CMakeLists.txt [hunk 4]
```diff
@@ -33,16 +33,21 @@ if(BUILD_TESTS)
 include_directories (
     ${GMOCK_INCLUDE_DIRECTORY}
     ${COMPONENTS_DIR}/hmi_message_handler/include
+    ${COMPONENTS_DIR}/hmi_message_handler/test/include
```
Comment (Kozoriz) [inline src/components/hmi_message_handler/test/CMakeLists.txt]: you can just write "include" instead this long line

**Comment by Kozoriz** [comment]
reviewed

**Comment by iAndrew5** [comment]
Reviewed

**Comment by VVeremjova** [comment]
Reviewed

File: src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  EXPECT_CALL(*mock_hmi_message_observer_, OnErrorSending(empty_message));
+  // Act
+  hmi_handler_->OnErrorSending(empty_message);
+}
+
+TEST_F(HMIMessageHandlerImplTest,
+       OnErrorSending_NotEmptyMessage_ExpectOnErrorSendingProceeded) {
+  // Arrange
+  utils::SharedPtr<application_manager::Message> message(
+      new application_manager::Message(
```
Comment (AGaliuzov) [inline src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc]: use `MakeShared` here

**Comment by anosach-luxoft** [comment]
@OHerasym, wrong branch name. Should be `hotfix`

**Comment by OHerasym** [comment]
@anosach-luxoft , There was incorect tittle, branch name is correct.

**Pull request merged**